extensions/slack-status/src/actions.tsx

```diff
@@ -0,0 +1,383 @@
+import {
+  AI,
+  Action,
+  ActionPanel,
+  Icon,
+  Keyboard,
+  Toast,
+  clearSearchBar,
+  confirmAlert,
+  showToast,
+  useNavigation,
+} from "@raycast/api";
+import { MutatePromise } from "@raycast/utils";
+import { Profile } from "@slack/web-api/dist/response/UsersProfileGetResponse";
+import { durationTitleMap } from "./durations";
+import { getCodeForEmoji, getEmojiForCode } from "./emojis";
+import { StatusForm } from "./form";
+import { useSlack } from "./slack";
+import { SlackStatusPreset } from "./types";
+import { showToastWithPromise } from "./utils";
+
+// Status Actions
+
+export function ClearStatusAction(props: { mutate: MutatePromise<Profile | undefined> }) {
+  const slack = useSlack();
+  return (
+    <Action
+      title="Clear Status"
+      icon={Icon.XMarkCircle}
+      onAction={async () => {
+        await showToastWithPromise(
+          props.mutate(
+            slack.users.profile.set({
+              profile: JSON.stringify({
+                status_text: "",
+                status_expiration: 0,
+                status_emoji: "",
+              }),
+            }),
+            {
+              optimisticUpdate() {
+                return {};
+              },
+            },
+          ),
+          {
+            loading: "Clearing status...",
+            success: "Cleared status",
+            error: "Failed clearing status",
+          },
+        );
+      }}
+    />
+  );
+}
+
+export function SetStatusWithAIAction(props: { statusText: string; mutate: MutatePromise<Profile | undefined> }) {
+  const slack = useSlack();
+  return (
+    <Action
+      title="Set Status"
+      icon={Icon.Stars}
+      onAction={async () => {
+        showToastWithPromise(
+          async () => {
+            const answer = await AI.ask(
+              `You help a Slack user set their status.
+              
+              Respond with the following JSON for the Slack status:
+              {
+                "text": <string for status text, should be short and sweet, no punctuation at the end, e.g. "Working out", "Listening to Drake's new album", "Coffe break">,
+                "emoji": <string for single emoji matching the text of the status>,
+                "duration": <optional integer for duration of status in seconds, only include if user specified duration or end of status in their description>
+              }
+
+              Current time of users status: ${new Date().toLocaleTimeString()}. User's description of their status: ${
```

Comment:
Should the user's description of their status be sent to the AI If it turns out to be empty?

Reply:
The action is only available if the `searchText` is not empty. Therefore the user's description will also not be empty.